### PR TITLE
Add more ndm:u functions.

### PR DESCRIPTION
**Pull request by joel16**
Added the following ndm:u (Network Daemon Manager) functions:

- Result NDMU_GetExclusiveState(NDM_ExclusiveState *state);
- Result NDMU_LockState(void);
- Result NDMU_UnlockState(void);
- Result NDMU_SuspendScheduler(u32 flag);
- Result NDMU_ResumeScheduler(void);
- Result NDMU_GetCurrentState(NDM_State *state);
- Result NDMU_QueryStatus(NDM_Daemon_Status *state);
- Result NDMU_SetScanInterval(u32 interval);
- Result NDMU_GetScanInterval(u32 *interval);
- Result NDMU_GetRetryInterval(u32 *interval);
- Result NDMU_ResetDaemons(void);
- Result NDMU_ClearMacFilter(void);

### CODE REVIEW

File: libctru/include/3ds/services/ndm.h [hunk 1]
```diff
@@ -4,21 +4,124 @@
...
 #pragma once
 
+/// Exclusive states.
 typedef enum {
 	EXCLUSIVE_STATE_NONE = 0,
 	EXCLUSIVE_STATE_INFRASTRUCTURE = 1,
 	EXCLUSIVE_STATE_LOCAL_COMMUNICATIONS = 2,
 	EXCLUSIVE_STATE_STREETPASS = 3,
-	EXCLUSIVE_STATE_STREETPASS_DATA = 4,
+	EXCLUSIVE_STATE_STREETPASS_DATA = 4
```
Comment (fincs) [inline libctru/include/3ds/services/ndm.h]: Don't remove this comma

File: libctru/include/3ds/services/ndm.h [hunk 2]
```diff
@@ -4,21 +4,124 @@
...
+	STATE_INFRASTRUCTURE_CONNECTING = 2,
+	STATE_INFRASTRUCTURE_CONNECTED = 3,
+	STATE_INFRASTRUCTURE_WORKING = 4,
+	STATE_INFRASTRUCTURE_SUSPENDING = 5,
+	STATE_INFRASTRUCTURE_FORCE_SUSPENDING = 6,
+	STATE_INFRASTRUCTURE_DISCONNECTING = 7,
+	STATE_INFRASTRUCTURE_FORCE_DISCONNECTING = 8,
+	STATE_CEC_WORKING = 9,
+	STATE_CEC_FORCE_SUSPENDING = 10,
+	STATE_CEC_SUSPENDING = 11
```
Comment (fincs) [inline libctru/include/3ds/services/ndm.h]: Add a comma here

File: libctru/include/3ds/services/ndm.h [hunk 3]
```diff
@@ -4,21 +4,124 @@
...
+} NDM_State;
+
+// Daemons.
+typedef enum {
+	DAEMON_CEC = 0,
+	DAEMON_BOSS = 1,
+	DAEMON_NIM = 2,
+	DAEMON_FRIENDS = 3
```
Comment (fincs) [inline libctru/include/3ds/services/ndm.h]: Add a comma here

File: libctru/include/3ds/services/ndm.h [hunk 4]
```diff
@@ -4,21 +4,124 @@
...
+} NDM_Daemon;
+
+/// Used to specify multiple daemons.
+typedef enum {
+	DAEMON_MASK_CEC = 1 << DAEMON_CEC,
+	DAEMON_MASK_BOSS = 1 << DAEMON_BOSS,
+	DAEMON_MASK_NIM = 1 << DAEMON_NIM,
+	DAEMON_MASK_FRIENDS = 1 << DAEMON_FRIENDS
```
Comment (fincs) [inline libctru/include/3ds/services/ndm.h]: Add a comma here

File: libctru/include/3ds/services/ndm.h [hunk 5]
```diff
@@ -4,21 +4,124 @@
...
+} NDM_Daemon_Mask;
+
+// Daemon status.
+typedef enum {
+	DAEMON_STATUS_BUSY = 0,
+	DAEMON_STATUS_IDLE = 1,
+	DAEMON_STATUS_SUSPENDING = 2,
+	DAEMON_STATUS_SUSPENDED = 3
+} NDM_Daemon_Status;
```
Comment (fincs) [inline libctru/include/3ds/services/ndm.h]: `ndmDaemonStatus`

File: libctru/include/3ds/services/ndm.h [hunk 6]
```diff
@@ -4,21 +4,124 @@
...
+	DAEMON_FRIENDS = 3
+} NDM_Daemon;
+
+/// Used to specify multiple daemons.
+typedef enum {
+	DAEMON_MASK_CEC = 1 << DAEMON_CEC,
+	DAEMON_MASK_BOSS = 1 << DAEMON_BOSS,
+	DAEMON_MASK_NIM = 1 << DAEMON_NIM,
+	DAEMON_MASK_FRIENDS = 1 << DAEMON_FRIENDS
+} NDM_Daemon_Mask;
```
Comment (fincs) [inline libctru/include/3ds/services/ndm.h]: `ndmDaemonMask`

File: libctru/include/3ds/services/ndm.h [hunk 7]
```diff
@@ -4,21 +4,124 @@
...
+	STATE_CEC_SUSPENDING = 11
+} NDM_State;
+
+// Daemons.
+typedef enum {
+	DAEMON_CEC = 0,
+	DAEMON_BOSS = 1,
+	DAEMON_NIM = 2,
+	DAEMON_FRIENDS = 3
+} NDM_Daemon;
```
Comment (fincs) [inline libctru/include/3ds/services/ndm.h]: `ndmDaemon`

File: libctru/include/3ds/services/ndm.h [hunk 8]
```diff
@@ -4,21 +4,124 @@
...
+	STATE_INFRASTRUCTURE_CONNECTED = 3,
+	STATE_INFRASTRUCTURE_WORKING = 4,
+	STATE_INFRASTRUCTURE_SUSPENDING = 5,
+	STATE_INFRASTRUCTURE_FORCE_SUSPENDING = 6,
+	STATE_INFRASTRUCTURE_DISCONNECTING = 7,
+	STATE_INFRASTRUCTURE_FORCE_DISCONNECTING = 8,
+	STATE_CEC_WORKING = 9,
+	STATE_CEC_FORCE_SUSPENDING = 10,
+	STATE_CEC_SUSPENDING = 11
+} NDM_State;
```
Comment (fincs) [inline libctru/include/3ds/services/ndm.h]: `ndmState`

File: libctru/include/3ds/services/ndm.h [hunk 9]
```diff
@@ -4,21 +4,124 @@
...
+typedef enum {
+	DAEMON_CEC = 0,
+	DAEMON_BOSS = 1,
+	DAEMON_NIM = 2,
+	DAEMON_FRIENDS = 3,
+} ndmDaemon;
+
+/// Used to specify multiple daemons.
+typedef enum {
+	DAEMON_MASK_CEC = 1 << DAEMON_CEC,
```
Comment (mtheall) [inline libctru/include/3ds/services/ndm.h]: I would use `DAEMON_MASK_CEC = BIT(DAEMON_CEC)` and similar for next lines. I would probably change the name to `DAEMON_CEC_BIT`.

Comment (joel16) [reply libctru/include/3ds/services/ndm.h]: Just to make sure, you want to DAEMON_CEC -> DAEMON_CEC_BIT?

Comment (mtheall) [reply libctru/include/3ds/services/ndm.h]: Logically I suppose either way could be argued. I might prefer `DAEMON_CEC_BIT = 0` and `DAEMON_CEC = BIT(DAEMON_CEC_BIT)`

**Pull request merged**